mempool/src/config.rs

```diff
@@ -46,6 +47,61 @@ impl std::fmt::Display for MempoolMaxSize {
     }
 }
 
+impl serde::Serialize for MempoolMaxSize {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.serialize_u64(self.0 as u64)
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+enum StringOrUint {
+    String(String),
+    UInt(u64),
+}
+
+impl<'de> serde::Deserialize<'de> for MempoolMaxSize {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::Error;
+
+        match StringOrUint::deserialize(deserializer)? {
+            StringOrUint::String(s) => {
+                let bytes: byte_unit::Byte = rpc::handle_result(byte_unit::Byte::from_str(&s))
+                    .map_err(|e| {
+                        D::Error::custom(format!(
+                            "Conversion to integer from data size failed: {e}"
+                        ))
+                    })?;
+
+                let bytes: usize = bytes.as_u64().try_into().map_err(|_| {
+                    D::Error::custom(format!("Integer {} is too large to fit", bytes))
+                })?;
+
+                let max_size = MempoolMaxSize::from_bytes(bytes);
+
+                Ok(max_size)
+            }
+            StringOrUint::UInt(bytes) => {
+                let bytes: usize = bytes.try_into().map_err(|_| {
+                    D::Error::custom(format!("Integer {} is too large to fit", bytes))
+                })?;
+
+                let max_size = MempoolMaxSize::from_bytes(bytes);
+
+                Ok(max_size)
+            }
+        }
```

Comment:
The `byte_unit::Byte` apparently already provides serde impls that support numbers and strings. The `serde` feature has to be enabled for it. Could that be reused?

https://docs.rs/byte-unit/latest/src/byte_unit/byte/serde_traits.rs.html#29-80